### ISSUE TITLE
LG-301: Add Liberty requestTiming metrics to /metrics

### ISSUE THREAD

**Issue by donbourne** [OpenLiberty/open-liberty#15781]
Add existing metrics from requestTiming MXBean to mpMetrics feature so they are exposed via /metrics

See list of metrics here:
https://www.ibm.com/support/knowledgecenter/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/rwlp_mon_request.html

**Post edit:**
The feature LG-301 introduces a new functionality for the existing `mpMetrics-2.x` and `mpMetricx-3.x` by adding new vendor metrics that gather data from the request timing monitor Mbean. These include the total `request count`, the current `active requests`, a `slow request count` and `hung request count`. This is the new functionality we are introducing and will be beta-ed for 21.0.0.6 beta and will be GA-ed for 21.0.0.6 It is currently beta-fenced.

However, to accomplish this the request timing monitor bundles and it's  related auto-feature had to be moved from WS-CD to Open Liberty. This also includes the MBean API. The feature in question `com.ibm.websphere.appserver.autoTimingMonitor-1.0.feature` is already GA-ed in WS-CD and when moved over to Open Liberty will also be in the GA state. This will be included in the shipment with 21.0.0.5. 

In Summary:
- Request Timing Monitor bundle + feature + API  are moved over from WS-CD and are shipped as GA for 21.0.0.5
- New vendor metrics using the Request Timing Monitor MBean is beta-fenced. Will be beta-d for 21.0.0.6-beta and GA-ed for 21.0.0.6.

As a result, this feature is labelled with target:21005 and target:21006-beta where the new functionality is in beta. However, approvals are still being seaked for 21.0.0.5.

##
#### Before proceeding to any items below (active development), this feature must be prioritized on the backlog, and have either the "Design Approved" or "No Design Approved" labels.  Follow the Feature and UFO Approval Process.
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## When complete & mandatory, add links to the UFO (Upcoming Feature Overview) document, FTS (Feature Test Summary), and blogs post issues(s):
- UFO:
- FTS:
- Beta Blog Post (if applicable):
- Blog Post: 

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)

Instructions:
- Do the actions below and mark them complete in the checklist when they are done.
- Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.

## **Design**
Before Development Starts or 8 weeks before Onboarding
- [ ] POC Design / UFO Review Scheduled (David Chang) or N/A.
- [ ] POC Design / UFO Reviewed (Feature Owner) or N/A.
- [ ] Complete any follow-ons from the POC Review.
- [ ] Design / UFO Approval ([Alasdair Nottingham](https://github.com/NottyCode)) or N/A.
- [x] No Design / No UFO Approval ([Arthur De Magalhaes](https://github.com/arthurdm) - cloud / [Alasdair Nottingham](https://github.com/NottyCode) - server) or N/A.
- [ ] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [ ] ID Requirements identified ([Documenting Open Liberty](https://github.com/OpenLiberty/open-liberty/wiki/Documenting-Open-Liberty)). (Epic owner / Feature owner with ID focal point)
- [ ] Create a child task of this epic entitled "Feature Test Summary" via [this template](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=&labels=Feature+Test+Summary&template=feature_test_summary.md&title=). Add the link in above.

## **Beta**
If your feature, or portions of it, are going to be included in a beta  
Before Onboarding the beta
- [ ] Beta Fence the functionality (`kind=beta`, `ibm:beta`, `ProductInfo.getBetaEdition()`)  

1 week before beta GA
- [ ] Create, populate, and link to the [Beta blog post issue](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=lauracowen%2C+jakub-pomykala&labels=&template=blog_post_beta.md&title=BETA+BLOG+-+title_of_your_update)

## **Legal**
3 weeks before Onboarding
- [x] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.

## **Translation**
3 weeks before Onboarding
- [ ] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)

## **Feature Complete**
2 weeks before Onboarding
- [x] Implementation complete. (Epic owner / Feature owner)
- [x] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [ ] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)

## **Focal Point Approvals**
2 to 1 week before Onboarding
#### You **MUST** have the Design Approved or No Design Approved label before requesting focal point approvals.

All features (both "Design Approved" and "No Design Approved")
- [ ] **FAT** - ([OpenLiberty/fat-approvers](https://github.com/orgs/OpenLiberty/teams/fat-approvers)). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [ ] **Demo** - ([Tom Evans](https://github.com/tevans78) or [Chuck Bridgham](https://github.com/cbridgha)). Demo is scheduled for an upcoming EOI. Approver adds label focalApproved:demo to the Epic in Github.
- [ ] **Globalization** ([Sam Wong](https://github.com/samwatibm) - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.

"Design Approved" features
- [ ] **Accessibility** - ([Steven Zvonek](https://github.com/steven1046)). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [ ] **ID** - ([Karen Deen](https://github.com/chirp1)). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [ ] **Performance** - ([Jared Anderson](https://github.com/jhanders34)). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [ ] **Serviceability** - ([Don Bourne](https://github.com/donbourne)). Serviceability has been addressed.
- [ ] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [ ] **SVT** - ([Brian Hanczaryk](https://github.com/hanczaryk) - APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.

## **Ready for GA**
1 week before Onboarding
- [ ] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the release branch. (Epic owner / Feature owner / Backlog Subtribe PM)

1 week before GA
- [ ] Create, populate, and link to the [Blog post issue](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=lauracowen%2C+jakub-pomykala&labels=&template=blog_post_ga_release.md&title=GA+BLOG+-+title_of_your_update)

## **Other deliverbles**
- [ ] **OL Guides** - ([Yee-Kang Chang](https://github.com/yeekangc)). Assessment for OL Guides is complete or N/A.
- [ ] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
- [ ] **Blog** - ([Laura Cowen](https://github.com/lauracowen)) Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by Channyboy** [comment]
Request Timing monitor bundle/features have been moved over from WS-CD.

The "new" functionality that adds the request timing monitor metrics to mpMetrics-2.x and mpMetrix-3.x is beta fenced with:
Beta Fenced with : https://github.com/OpenLiberty/open-liberty/pull/16623

**Comment by Channyboy** [comment]
Feature Test summary at : https://github.com/OpenLiberty/open-liberty/issues/16709

**Comment by Channyboy** [comment]
^ @OpenLiberty/fat-approvers

**Comment by Channyboy** [comment]
@samwatibm I'd like to request a globalization request for this epic. The request for globalization was requested last week and am awaiting return on the shipment.

**Comment by Channyboy** [comment]
The feature LG-301 introduces a new functionality for the existing `mpMetrics-2.x` and `mpMetricx-3.x` by adding new vendor metrics that gather data from the request timing monitor Mbean. These include the total `request count`, the current `active requests`, a `slow request count` and `hung request count`. This is the new functionality we are introducing and will be beta-ed for 21.0.0.6 beta and will be GA-ed for 21.0.0.6 It is currently beta-fenced.

However, to accomplish this the request timing monitor bundles and it's  related auto-feature had to be moved from WS-CD to Open Liberty. This also includes the MBean API. The feature in question `com.ibm.websphere.appserver.autoTimingMonitor-1.0.feature` is already GA-ed in WS-CD and when moved over to Open Liberty will also be in the GA state. This will be included in the shipment with 21.0.0.5. 

In Summary:
- Request Timing Monitor bundle + feature + API  are moved over from WS-CD and are shipped as GA for 21.0.0.5
- New vendor metrics using the Request Timing Monitor MBean is beta-fenced. Will be beta-d for 21.0.0.6-beta and GA-ed for 21.0.0.6.

As a result, this feature is labelled with `target:21005` and `target:21006-beta` where the _new functionality_ is in beta. However, approvals are still being seaked for 21.0.0.5.

Fyi: @NottyCode @tevans78 @mbcapobia 
If this is not right approval process for this situation, please advise.

**Comment by samwatibm** [comment]
@Channyboy Sorry but if translations were requested last week, the 21.0.0.5 translations aren't back yet.   Approvals aren't granted until the translations are back.

**Comment by donbourne** [comment]
The globalization relates to the beta portion of this epic -- so I think technically we don't need that approved until 21.0.0.6 GA timeframe.

The key thing for right now is to ensure @NottyCode and @tevans78 are ok with the move of the request timing monitor bundle/feature/api to OL, which won't be beta-guarded and will go straight to GA (since it's hard to beta-guard open-sourcing a feature).

**Comment by Channyboy** [comment]
@samwatibm  
Translation have been returned here https://github.com/OpenLiberty/open-liberty/pull/16792

**Comment by samwatibm** [comment]
Waiting for translations to be merged.  They are now

**Comment by donbourne** [comment]
@Channyboy , can we do a beta blog to announce the coming new "prometheus" metrics for requestTiming?

**Comment by Channyboy** [comment]
egaed with 21006
